apis/projectcontour/v1/httpproxy.go

```diff
@@ -684,6 +688,92 @@ type HTTPDirectResponsePolicy struct {
 	Body string `json:"body,omitempty"`
 }
 
+// StatusCodeType defines the type of status code match in HTTPResponseOverridePolicy
+// +kubebuilder:validation:Enum=Value;Range
+type StatusCodeType string
+
+// StatusCodeValue defines a single status code to match
+type StatusCodeValue struct {
+	// Value is the exact status code to match
+	// +kubebuilder:validation:Minimum=100
+	// +kubebuilder:validation:Maximum=599
+	// +required
+	Value int `json:"value"`
+}
+
+// StatusCodeRange defines a range of status codes to match
+type StatusCodeRange struct {
+	// Start is the beginning of the status code range (inclusive)
+	// +kubebuilder:validation:Minimum=100
+	// +kubebuilder:validation:Maximum=599
+	// +required
+	Start int `json:"start"`
+
+	// End is the end of the status code range (inclusive)
+	// +kubebuilder:validation:Minimum=100
+	// +kubebuilder:validation:Maximum=599
+	// +required
+	End int `json:"end"`
+}
+
+// StatusCodeMatch defines a status code match condition
+type StatusCodeMatch struct {
+	// Type specifies how to match the status code
+	// +required
+	Type StatusCodeType `json:"type"`
+
+	// Value is used when Type is Value to specify an exact status code
+	// +optional
+	Value int `json:"value,omitempty"`
+
+	// Range is used when Type is Range to specify a range of status codes
+	// +optional
+	Range *StatusCodeRange `json:"range,omitempty"`
+}
+
+// ResponseBodyConfig defines the response body to use in response override
+type ResponseBodyConfig struct {
+	// Type specifies the type of body content to be used
+	// +kubebuilder:validation:Enum=Inline
+	// +required
+	Type string `json:"type"`
+
+	// Inline provides the content directly in the HTTPProxy resource
+	// +optional
```

Comment:
we should add a size limit here to make sure huge response page content is not configured

Reply:
Good point here. I've added a 4KB limit.